MaxText/max_utils.py

```diff
@@ -348,7 +348,10 @@ def init_initial_state(model, tx, config, is_training, key):
                           jnp.ones(input_shape, dtype=jnp.int32),
                           jnp.ones(input_shape, dtype=jnp.int32))
   if is_training:
-    return init_training_state(model.apply, model_vars['params'], tx)
+    if config.quantization=="fp8":
+      return init_training_state(model.apply, model_vars, tx)
```

Comment:
This is very suspicious to me. Why is it necessary, why can't we find a more universal solution?

Reply:
> Ah is this because the flax linen impl adds those variables in the overwrite_with_gradients namespace?

@rwitten See the resolved conversation below. `fp8ops` has the "hack" to have some variables keyed under `overwrite_with_gradients` so they can be manually overwritten, which leads to this weirdness. But agree - its odd. Maybe there is a better way to do this...

Reply:
@abhinavgoel95 Hmm what happens if we just pass in `model_vars` always here? Looks like before in MaxText we were extracting the params directly, but Flax should be able to just infer the direct parameter collection as being equivalent to `{'params': params}`

Reply:
This is needed by the Fp8 implementation in Flax.

Reply:
cc @kaixih